finagle/buoyant/src/main/scala/com/twitter/finagle/buoyant/SocketOptionsConfig.scala

```diff
@@ -2,13 +2,30 @@ package com.twitter.finagle.buoyant
 
 import com.twitter.finagle.Stack
 import com.twitter.finagle.transport.Transport
+import com.twitter.util.Duration
 
 case class SocketOptionsConfig(
   noDelay: Boolean = true,
   reuseAddr: Boolean = true,
-  reusePort: Boolean = false
+  reusePort: Boolean = false,
+  writeTimeoutMs: Option[Long] = None,
+  readTimeoutMs: Option[Long] = None,
+  keepAlive: Option[Boolean] = None
 ) {
-  def params = Stack.Params.empty +
-    Transport.Options(noDelay, reuseAddr, reusePort)
+  def params: Stack.Params = {
+    val writeTimeout: Duration = writeTimeoutMs match {
+      case Some(seconds) => Duration.fromMilliseconds(seconds)
```

Comment:
nit: probably call this variable `milliseconds` instead of `seconds`.

Reply:
done